server/src-exec/Migrate.hs

```diff
@@ -15,421 +23,101 @@ import           Hasura.Server.Query
-  liftTx $ Q.catchE defaultTxErrorHandler $
-  Q.multiQ [Q.sql|
-            UPDATE hdb_catalog.hdb_table
-            SET is_system_defined = 'true'
-            WHERE table_schema = 'hdb_catalog'
-             AND  table_name = 'hdb_allowlist';
-           |]
-
-from17To18 :: MonadTx m => m ()
-from17To18 =
-  liftTx $ Q.catchE defaultTxErrorHandler $
-  Q.multiQ [Q.sql|
-            DELETE FROM hdb_catalog.hdb_table
-            WHERE table_schema = 'hdb_catalog'
-              AND table_name = 'hdb_query_template';
-            DROP table hdb_catalog.hdb_query_template
-           |]
-
-from18To19 :: MonadTx m => m ()
-from18To19 = do
-  -- Migrate database
-  Q.Discard () <- liftTx $ Q.multiQE defaultTxErrorHandler
-    $(Q.sqlFromFile "src-rsr/migrate_from_18_to_19.sql")
-  return ()
-
-from19To20 :: (MonadTx m) => m ()
-from19To20 = do
-  Q.Discard () <- liftTx $ Q.multiQE defaultTxErrorHandler
-    $(Q.sqlFromFile "src-rsr/migrate_from_19_to_20.sql")
-  pure ()
+    -- the old 0.8 catalog version is non-integral, so we store it in the database as a string
+    latestCatalogVersionString = T.pack $ show latestCatalogVersion
 
-from20To21 :: (MonadTx m) => m ()
-from20To21 = liftTx $ Q.catchE defaultTxErrorHandler $ do
-  Q.unitQ "CREATE INDEX ON hdb_catalog.event_log (locked)" () False
+    getCatalogVersion = liftTx $ runIdentity . Q.getRow <$> Q.withQE defaultTxErrorHandler
+      [Q.sql| SELECT version FROM hdb_catalog.hdb_version |] () False
 
-from21To22 :: (MonadTx m) => m ()
-from21To22 = do
-  Q.Discard () <- liftTx $ Q.multiQE defaultTxErrorHandler
-    $(Q.sqlFromFile "src-rsr/migrate_from_21_to_22.sql")
-  pure ()
-
-from22To23 :: (MonadTx m) => m ()
-from22To23 = do
-  Q.Discard () <- liftTx $ Q.multiQE defaultTxErrorHandler
-    $(Q.sqlFromFile "src-rsr/migrate_from_22_to_23.sql")
-  pure ()
-
-from23To24 :: MonadTx m => m ()
-from23To24 =
-  liftTx $ Q.catchE defaultTxErrorHandler $
-  Q.multiQ [Q.sql|
-            ALTER TABLE hdb_catalog.hdb_table
-            ADD COLUMN configuration JSONB NOT NULL DEFAULT '{}'::jsonb;
-           |]
-migrateCatalog
-  :: ( MonadTx m
-     , CacheRWM m
-     , MonadIO m
-     , UserInfoM m
-     , HasHttpManager m
-     , HasSQLGenCtx m
-     )
-  => UTCTime -> m String
-migrateCatalog migrationTime = migrateFrom =<< getCatalogVersion
-  where
+    migrateFrom :: T.Text -> m T.Text
     migrateFrom previousVersion
-      | previousVersion == curCatalogVer =
-          return $ "already at the latest version. current version: " <> show curCatalogVer
-      | [] <- neededMigrations =
-          throw400 NotSupported $ "unsupported version : " <> previousVersion
-      | otherwise =
-          traverse_ snd neededMigrations >> postMigrate
+      | previousVersion == latestCatalogVersionString = pure $
+          "already at the latest version. current version: " <> latestCatalogVersionString
+      | [] <- neededMigrations = throw400 NotSupported $ "unsupported version : " <> previousVersion
```

Comment:
Always hated this error message. Can we change it to something which makes more sense for our user?

Reply:
Good call, I’ve modified the message.